cranelift/codegen/src/isa/aarch64/lower.rs

```diff
@@ -1252,14 +1279,14 @@ pub(crate) fn lower_icmp<C: LowerCtx<I = Inst>>(
     ctx: &mut C,
     insn: IRInst,
     condcode: IntCC,
-    output: IcmpOutput,
-) -> CodegenResult<()> {
+    target: IcmpTarget,
+) -> CodegenResult<IcmpResult> {
```

Comment:
I'm not 100% happy with how this turned out.

My issue is that this signature implies that we can change the output register, despite us never doing that. However changing `IcmpResult` into something like:

```
enum IcmpResult {
    Flag(IntCC),
    Register,
}
```

Also doesn't seem great, because now the output is inconsistent: if you request flag outputs you have to check the result and if you request register outputs you can ignore the result.

Do you guys have any suggestion on how to improve this?


This PR is also getting pretty big on the AArch64 side, if you would like me to separate it into 2 PR's let me know!